extended/src/main/java/io/kubernetes/client/extended/network/RoundRobinLoadBalanceStrategy.java

```diff
@@ -0,0 +1,38 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package io.kubernetes.client.extended.network;
+
+import java.util.List;
+import java.util.concurrent.locks.ReentrantLock;
+
+public class RoundRobinLoadBalanceStrategy implements LoadBalanceStrategy {
+
+  private final ReentrantLock reentrantLock = new ReentrantLock();
+
+  private int robinIndex = 0;
+
+  @Override
+  public String chooseIP(List<String> availableIPs) {
+    if (availableIPs == null || availableIPs.size() == 0) {
+      throw new IllegalArgumentException("failed choosing IP target: empty candidates");
+    }
+    reentrantLock.lock();
```

Comment:
Would it be better to use `AtomicInteger` this lock seems heavy-weight.

Reply:
Also, though this is "round robin" I think you could actually just choose randomly and you'd get the same behavior, I don't think that round robin requires a deterministic order.

Reply:
> Would it be better to use AtomicInteger this lock seems heavy-weight.

updated

> Also, though this is "round robin" I think you could actually just choose randomly and you'd get the same behavior, I don't think that round robin requires a deterministic order.

i presume the randomized strategy you're saying is a bit different from the proposing one here. for the cases where the client is raising requests at a lower rate e.g. "watching" requests in one loop. the proposed "round robin" in this pull will make sure the even distribution in order, while if we're doing a randomized "round robin", there's a chance of uneven distribution for lower-rate client, is it?